### ISSUE TITLE
link-in-text-block should pass if sufficient contrast ratio even without distinct style

### ISSUE THREAD

**Issue by LiLoDavis** [dequelabs/axe-core#2331]
**Expectation**: Rule should **pass** if a link has a contrast ratio of at least 3:1 with surrounding text even if it has no distinct style

**Actual**: Rule returns **uncertain**.

**Motivation**: WCAG Success Criterion requires sufficient contrast OR distinct style, not both

Excerpt from WCAG F73: F[ailure of Success Criterion 1.4.1 due to creating links that are not visually evident without color vision](https://www.w3.org/WAI/WCAG21/Techniques/failures/F73):

> Red and Pink are the same color (hue) but they have different lightness (which is not color). So red and pink would pass the requirement for "not distinguished by color (hue) alone" since they differ by lightness (which is not color) - as long as the difference in lightness (contrast) is 3:1 or greater. 

The test procedure also makes it clear that a link with sufficient contrast passes this SC:

> Check that each link in the page that is identifiable by color (hue) is visually identifiable via some other means (e.g., underlined, bolded, italicized, sufficient difference in lightness, etc.)

Also, there are some problems with accompanying messages for uncertain outcomes: 

**Problem**: the uncertain message says the background color could not be determined, when that's not true.
For example, when the specified link color provides sufficient contrast, setting the text-decoration to none triggers this message: 

> Element's contrast ratio could not be determined. Check for a distinct hover/focus style.

**Problem**: the uncertain message sometimes says to check for a distinct hover/focus style, but that has no impact on pass/fail outcome.
It's true that WCAG Technique [G183: Using a contrast ratio of 3:1 with surrounding text and providing additional visual cues on focus for links or controls where color alone is used to identify them](https://www.w3.org/WAI/WCAG21/Techniques/general/G183) encourages using additional visual cues, but sufficient techniques are informative, not required.

<pre><code>
axe-core version: 3.5.2-canary37be85f 
in axe coconut v4.4.1

**Comment by WilcoFiers** [comment]
I'm going to have to look into this. I've always understood this to include hover / focus effect, but you've got a very good point here.

**Comment by dylanb** [comment]
@WilcoFiers this appears to be a duplicate of https://github.com/dequelabs/axe-core/issues/2817
